bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/VisitDocumentController.java

```diff
@@ -61,16 +63,25 @@ public VisitDocumentResponse save(@RequestBody VisitDocumentRequest visitDocumen
     @RequestMapping(method = RequestMethod.POST, value = baseVisitDocumentUrl + "/uploadDocument")
     @ResponseBody
     public HashMap<String, String> saveDocument(@RequestBody Document document) {
-        Patient patient = Context.getPatientService().getPatientByUuid(document.getPatientUuid());
-        String encounterTypeName = document.getEncounterTypeName();
-        if (StringUtils.isEmpty(encounterTypeName)) {
-            encounterTypeName = administrationService.getGlobalProperty("bahmni.encounterType.default");
+        try {
+            HashMap<String, String> savedDocument = new HashMap<>();
+            Patient patient = Context.getPatientService().getPatientByUuid(document.getPatientUuid());
+            String encounterTypeName = document.getEncounterTypeName();
+            if (StringUtils.isEmpty(encounterTypeName)) {
+                encounterTypeName = administrationService.getGlobalProperty("bahmni.encounterType.default");
+            }
+            String fileName = sanitizeFileName(document.getFileName());
+            Paths.get(fileName);
+
+            // Old files will follow: patientid-encounterName-uuid.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75.png)
+            // New ones will follow: patientid_encounterName_uuid__filename.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75__doc1.png)
```

Comment:
The comment says underscore will be used as delimiter, but example seems to use hyphens.

Reply:
In old style, generated file name will have hyphens as separator and in new style in end __originalFileName.ext will appended